src/db/connection.rs

```diff
@@ -23,12 +47,22 @@ impl From<Connection> for RequestContext {
 impl RequestContext {
     /// Get a DatabaseContext from this struct that can be used to connect
     /// to the db
-    pub fn database_context(
-        &self,
+    pub fn database_context<'a, 'b>(
+        &'b self,
+        resource: &'a str,
+        token: Option<Token>,
         action: ActionType,
         modifier: ActionModifier,
-    ) -> DatabaseContext {
-        return DatabaseContext::from(&self.conn, action, modifier);
+    ) -> Result<DatabaseContext<'a>>
+    where
+        'b: 'a,
```

Comment:
Is there a reason we're not changing the 'b to be 'a?

Reply:
Fixed!